Drivers/rc_receiver/Tests/sbus_gtest.cpp

```diff
@@ -0,0 +1,98 @@
+#include <gtest/gtest.h>
+#include <fff.h>
+#include "sbus_receiver.hpp"
+
+DEFINE_FFF_GLOBALS;
+
+/*
+    Faking the HAL function in this unit test to do nothing
+    The recieved data will be manually added to test cases
+*/
+FAKE_VALUE_FUNC(HAL_StatusTypeDef, HAL_UART_Receive_DMA, UART_HandleTypeDef*, uint8_t*, uint16_t );
+
+TEST (SBUS_Receiver_Test, RawDataToSBus){
+    const uint8_t in_bytes[SBUS_FRAME_SIZE] = {
+        HEADER_, 0x80, 0x08, 0x84,
+        0x20, 0x06, 0x41, 0x88, 0x42, 0x18,
+        0xE2, 0x10, 0x88, 0x48, 0x84, 0x22,
+        0x16, 0xC1, 0x88, 0x46, 0x38, 0xE2,
+        0x11, 0x00, FOOTER_
+    };
+
+    SBus_t packet{
+        packet.ch[0] = 128,
+        packet.ch[1] = 129,
+        packet.ch[2] = 130,
+        packet.ch[3] = 131,
+        packet.ch[4] = 132,
+        packet.ch[5] = 133,
+        packet.ch[6] = 134,
+        packet.ch[7] = 135,
+        packet.ch[8] = 136,
+        packet.ch[9] = 137,
+        packet.ch[10] = 138,
+        packet.ch[11] = 139,
+        packet.ch[12] = 140,
+        packet.ch[13] = 141,
+        packet.ch[14] = 142,
+        packet.ch[15] = 143
+    };
+
+    UART_HandleTypeDef uart_fake;
+    SBUSReceiver sbus_test(&uart_fake);
+    memcpy(sbus_test.raw_sbus_, in_bytes, sizeof(in_bytes));
```

Comment:
I'm surprised that `raw_sbus_` is public... this should probably be private, and you make use of the fake `HAL_UART_Receive_DMA` to get the address to memcpy into.

Reply:
I made raw_sbus_ to be public because of the nature of DMA callback. The DMA function takes a buffer pointer as a parameter and updates the value when the DMA interrupt was triggered. Also in the callback function, I need to call DMA again to make sure it runs circularly.  Maybe I can change the implementation by adding a getter method that returns the pointer of the buffer, so outside of the scope, the callback function could still modify the content of the buffer.